secure.c

```diff
@@ -901,7 +912,7 @@ sec_connect(char *server, char *username, char *domain, char *password, RD_BOOL
 		return False;
 
 	/* We exchange some RDP data during the MCS-Connect */
-	mcs_data.size = 512;
+	mcs_data.size = 542;
```

Comment:
Yes. That is the size of the head buffer, and it got larger in the protocol change enabling sending DPI.